### PR TITLE
Update settings to the new layout

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1202636062710671/f

### Description
Updates the Settings screen to match the latest from design. Including changes to:

- Groups (**general** and **other** are deleted, **appearance**, **customize** and **about** are added)
- Ordering

#### Ensure device is **enabled** for autofill (e.g., use `internal` build type)
- [x] Visit settings from overflow
- [x] Verify the layout matches the Figma (linked in Asana task)
- [x] Verify that autofill is present

#### Ensure device is **disabled** for autofill (e.g., use `play` build type)
- [x] Visit settings from overflow
- [x] Verify the layout matches the Figma (linked in Asana task)
- [x] Verify that autofill is omitted (and the dividers aren't messed up where autofill would have been displayed)

### CODE REVIEW

**Comment by CDRussell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2239** <a href="https://app.graphite.dev/github/pr/duckduckgo/android/2239" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/android/2239?utm_source=stack-comment).

File: app/src/main/res/layout/content_settings_autofill.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <TextView
+        android:id="@+id/autofill"
+        style="@style/SettingsItemClickable"
+        android:gravity="center_vertical"
+        android:text="@string/autofillSettingsDisplayName" />
+
+    <com.duckduckgo.mobile.android.ui.view.SectionDivider
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingBottom="0dp" />
```
Comment (CDRussell) [inline app/src/main/res/layout/content_settings_autofill.xml:32]: Padding reduction is a change from the last PR

**Review by PIYUSH-MISHRA-00 — COMMENTED** [review]
Done

**Comment by CDRussell** [comment]
@karlenDimla as noted previously, this PR is largely the same as the previous PR for the settings screen, except for the following changes:
- changes to margins / paddings (namely, in common-ui/src/main/res/values/styles.xml)
- reverting the copy change (so "automatically burn" is back as before as "automatically clear")

As before, it can't be merged until product review is finished. but can you re-review it so that part is ready to go, please?

**Review by karlenDimla — APPROVED** [review]
Looks good!

**Pull request merged**